### ISSUE TITLE
Unable to add Home Assistant

### ISSUE THREAD

**Issue by paulharman** [xpipe-io/xpipe#154]
Unable to add Home Assistant using the SSH addon - https://community.home-assistant.io/t/home-assistant-community-add-on-ssh-web-terminal/33820

Error:

Shell opener command was unsuccessful:
Corrupted MAC on input.
ssh_dispatch_run_fatal: Connection to 192.168.0.13 port 22: message authentication code incorrect

When logging in via SolarPutty, you get this screen on successful connection

![image](https://github.com/xpipe-io/xpipe/assets/3407038/d82dab92-3653-49d0-ba3d-ebd48c6393ca)

**Comment by crschnick** [comment]
How exactly do you connect to this? What happens if you run `ssh -T user@host` manually in the terminal?

**Comment by paulharman** [comment]
I just have a session in SolarPutty

![image](https://github.com/xpipe-io/xpipe/assets/3407038/8aacede0-e501-41e4-8af3-1991e66f47cc)

If I run `ssh -T alc@192.168.0.13` I get 

```
Corrupted MAC on input.
ssh_dispatch_run_fatal: Connection to 192.168.0.13 port 22: message authentication code incorrect
```

However, if I run `ssh -m hmac-sha2-256-etm@openssh.com alc@192.168.0.13` it connects correctly.

![image](https://github.com/xpipe-io/xpipe/assets/3407038/316fe4b9-f224-4829-8342-b95e0659edab)

**Comment by crschnick** [comment]
For now, since you can't add custom options to SSH connections in xpipe, you could add this connection to your user openssh config file at ~/.ssh/config with the appropriate options and then import that in xpipe.

**Comment by paulharman** [comment]
How can I do this in windows please?

**Comment by crschnick** [comment]
That should work exactly the same, the directory is `%USERPROFILE%\.ssh\config` for the OpenSSH config.

You should just be able to add an entry like this:

```
Host test_system
    HostName localhost
    User username
    Port 22
    MACs hmac-sha2-256-etm@openssh.com
```

and any other options that you need. You can follow a guide like this: https://linuxize.com/post/using-the-ssh-config-file/

**Comment by lubeda** [comment]
The ssh-config trick works, but is lacking all the bonus features that makes xpipe so helpful, especially the docker integration

**Comment by crschnick** [comment]
What exactly is not working for you? There shouldn't be a difference in functionality in both cases

**Comment by lubeda** [comment]
The ssh-addon in home assistant was recently [updated](https://github.com/hassio-addons/addon-ssh/releases). Before the latest update, I was able to at least see the docker containers running in home assistant. Now homeassistant (via ssh-config) is "only" kind of subfolder of my local Windows machine.

![screenshot of my windows machine](https://github.com/xpipe-io/xpipe/assets/17859549/7cd38d9d-caa4-47c4-9390-fe3e16a368c0)

**Comment by crschnick** [comment]
Ok, so I'm not familiar with home assistant and the ssh-addon and what issues it causes.

So your problem is that it does not find your docker containers when you search for them in the ssh config connection in xpipe here?
![image](https://github.com/xpipe-io/xpipe/assets/72509152/9bbd7a6a-7fe5-44b5-a451-1d91d9d28df3)

**Comment by lubeda** [comment]
Upps, sorry! I didn't press the search button 😱

**Comment by crschnick** [comment]
Alright now that the feature of customized ssh connections has been released in 1.7.15, I think we can close this.
